c7n/provider.py

```diff
@@ -73,22 +77,29 @@ def import_resource_classes(resource_map, resource_types):
 
     mod_map = {}
     rmods = set()
-    not_found = []
+    not_found = set()
+    found = []
 
     for r in resource_types:
         if r not in resource_map:
-            not_found.append(r)
+            not_found.add(r)
             continue
         rmodule, rclass = resource_map[r].rsplit('.', 1)
         rmods.add(rmodule)
 
     for rmodule in rmods:
         mod_map[rmodule] = importlib.import_module(rmodule)
 
-    return [getattr(mod_map[rmodule], rclass, None) for
-            rmodule, rclass in [
-                resource_map[r].rsplit('.', 1) for r in resource_types
-                if r in resource_map]], not_found
+    for rtype in resource_types:
+        if rtype in not_found:
+            continue
+        rmodule, rclass = resource_map[rtype].rsplit('.', 1)
+        r = getattr(mod_map[rmodule], rclass, None)
+        if r is None:
+            not_found.append(rtype)
```

Comment:
I believe this should be `.add()` for a `set` type.

Reply:
yup, fixed, thanks!